### PR TITLE
DRAFT: Introducing exportToFile to reduce duplicate codes

### PR DESCRIPTION
**Pull request by jinevening**
This introduces exportToFile to reduce duplicate codes

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to #3667

### CODE REVIEW

**Comment by jinevening** [comment]
**For reviewers**:  Please review `CircleExporter.h` and `CircleExporter.cpp`, which have the implementation of `exportToFile`.

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 1]
```diff
@@ -185,9 +184,7 @@ void RecordMinMax::saveModel(const std::string &output_model_path)
 {
   // Export to output Circle file
   luci::CircleExporter exporter;
-  CircleExpContract contract(_module.get(), output_model_path);
-
-  if (!exporter.invoke(&contract))
+  if (!exporter.exportToFile(_module.get(), output_model_path))
```
Comment (seanshpark) [inline compiler/record-minmax/src/RecordMinMax.cpp:187]: `CircleExpContract ` is introduced to separate and now you binded again.
This is not I hoped.

Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:187]: Maybe I misundertood your comment in https://github.com/Samsung/ONE/issues/3667#issuecomment-670206599.

Can you explain a little bit more?

Comment (seanshpark) [reply compiler/record-minmax/src/RecordMinMax.cpp:187]: I think the comment is somwhat misleading (I forgot somepart of the design)
`CircleExporter` which generates export data and storing the data was decoupled by `CircleExpContract`.
This was by design.
Misleading part is that comment says storing the data to file can be included in `CircleExporter` module which I didn't think is good thing todo.

What I think is exporting to file by some methould should be independent of `CircleExporter`.
`CircleExpContract` just calls that method.

Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:187]: @seanshpark  How about making `CircleFileExpContract.h` under the luci? Then, the final code would look like below.

```
#include <luci/CircleExporter.h>
#include <luci/CircleFileExpContract.h>
...

luci::CircleExporter exporter; 
CircleFileExpContract contract(_module.get(), output_model_path);

  if (!exporter.invoke(&contract))
...
```

Comment (seanshpark) [reply compiler/record-minmax/src/RecordMinMax.cpp:187]: > How about making CircleFileExpContract.h under the luci? 

there are several modules in luci. which one or a new one?

Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:187]: > there are several modules in luci. which one or a new one?


Maybe under the `luci/export/include/luci`? (where `CircleExporter.h` resides)

Comment (seanshpark) [reply compiler/record-minmax/src/RecordMinMax.cpp:187]: OK :) let's try

**Comment by jinevening** [comment]
@seanshpark I've changed the implementation to decouple exporting and saving. PTAL

**Comment by seanshpark** [comment]
Looks good :)

**Pull request closed without merging**